src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1351,6 +1352,25 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                                     builder.Add(BuildDirective());
                                     return;
                                 }
+                                break;
+                            case DirectiveTokenKind.Attribute:
+                                if (At(SyntaxKind.LeftBracket))
+                                {
+                                    AcceptUntil(SyntaxKind.RightBracket, SyntaxKind.NewLine);
```

Comment:
Is it possible to have nested brackets in an attribute signature?

Reply:
I can't think of a way other than within strings which is fine because it will be tokenized as StringLiteral.

Reply:
It is possible to have an attribute take in an array. And the attribute usage can look like `[custom(new[] {1,2,3})]`. I'll handle this logic.